app/main.py

```diff
@@ -1,3 +1,56 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Union["Distance", int, float]) -> "Distance":
+        if not isinstance(other, Distance):
+            self.km += other
+            return self
+
+        self.km += other.km
+        return self
+
+    def __mul__(self, multiplier: int | float) -> "Distance":
+        return Distance(self.km * multiplier)
+
+    def __truediv__(self, divisor: int | float) -> "Distance":
+        return Distance(round(self.km / divisor, 2))
+
+    def __eq__(self, other: Union["Distance", int, float]) -> bool:
+        if not isinstance(other, Distance):
+            return self.km == other
```

Comment:
Why don't u just remove `not` from this check and change places lines 36 and 38? It will make code more readable and easier to understand